### PR TITLE
Remove pycocotools dependency

### LINKED ISSUE #447: Remove pycocotools dependency from `fiftyone.utils.eval.evaluate_detections()`
We're only using pycocotools for a small computation in that library, and it would be better to just implement our own eval methods and get rid of pycocotools

### PR DESCRIPTION
**Pull request by ehofesmann**
## What changes are proposed in this pull request?

Reimplement the IoU calculation from pycocotools to remove the dependency on it. Removed all instances of pycocotools.

Closes #447 

## How is this patch tested? If it is not, in a single sentence please explain why.

Results from the new implementation matched the original one. The runtime was unchanged.

### What areas of FiftyOne does this PR affect?

-   [ ] `App`: FiftyOne application changes
-   [ ] `Build`: Build and test infrastructure changes
-   [x] `Core`: Core `fiftyone` Python library changes
-   [ ] `Documentation`: FiftyOne documentation changes
-   [ ] `Other`

### CODE REVIEW

File: docs/source/tutorials/evaluate_detections.ipynb [hunk 1]
```diff
@@ -25,7 +25,7 @@
     "\n",
     "In this tutorial, we'll use an off-the-shelf [Faster R-CNN detection model](https://pytorch.org/docs/stable/torchvision/models.html#faster-r-cnn) provided by PyTorch.\n",
     "\n",
-    "To use it, you'll need to install `torch` and `torchvision`, if necessary. We'll also need `pycocotools` for evaluation:"
+    "To use it, you'll need to install `torch` and `torchvision`, if necessary."
```
Comment (brimoor) [inline docs/source/tutorials/evaluate_detections.ipynb:28]: So `pycocotools` is not required for evaluation anymore, but please check if it is still required in order to download the COCO dataset from the zoo (per my comment later in the PR)

File: fiftyone/zoo/torch.py [hunk 2]
```diff
@@ -295,7 +295,6 @@ def supported_splits(self):
 
     def _download_and_prepare(self, dataset_dir, scratch_dir, split):
         def download_fcn(download_dir):
-            fou.ensure_pycocotools()
```
[the remark this answers is not in the document]
Comment (ehofesmann) [reply fiftyone/zoo/torch.py]: Reverted

File: docs/source/tutorials/evaluate_detections.ipynb (hunk 1 above)
Comment (ehofesmann) [reply docs/source/tutorials/evaluate_detections.ipynb]: Yeah, this is still required. My bad.

**Comment by ehofesmann** [comment]
@brimoor This is ready for rereview.

**Pull request merged**